### PR TITLE
[Feature] Add http trigger for dynamic partition

### PR DESCRIPTION
**Pull request by Astralidea**
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 5 / 44 (11.36%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/rest/TriggerAction.java|4|43|09.30%|[54, 56, 57, 58, 59, 63, 64, 66, 67, 70, 71, 73, 74, 75, 76, 79, 80, 81, 82, 83, 87, 88, 89, 90, 91, 93, 94, 97, 98, 99, 100, 103, 105, 106, 107, 108, 109, 110, 112]|
|:large_blue_circle:|com/starrocks/http/HttpServer.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 27 / 44 (61.36%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/rest/TriggerAction.java|26|43|60.47%|[57, 58, 59, 64, 66, 67, 74, 75, 76, 81, 82, 83, 98, 99, 100, 109, 110]|
|:large_blue_circle:|com/starrocks/http/HttpServer.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 39 / 44 (88.64%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/rest/TriggerAction.java|38|43|88.37%|[57, 58, 59, 109, 110]|
|:large_blue_circle:|com/starrocks/http/HttpServer.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

File: fe/fe-core/src/main/java/com/starrocks/http/rest/TriggerAction.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        String dbName = request.getSingleParameter(DB_PARAM);
+        String tableName = request.getSingleParameter(TABLE_PARAM);
+
+        if (Strings.isNullOrEmpty(dbName)) {
+            response.appendContent("Missing params. Need database name");
+            writeResponse(request, response, HttpResponseStatus.BAD_REQUEST);
+            return;
+        }
+
+        Database db = GlobalStateMgr.getCurrentState().getDb(dbName);
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/http/rest/TriggerAction.java:79]: is there any concurrency problems?

**Review by ABingHuang — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by Astralidea** [comment]
https://github.com/Mergifyio backport branch-3.2

**Comment by Astralidea** [comment]
https://github.com/Mergifyio backport branch-3.1

**Pull request merged**